src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1492,26 +1526,42 @@ tx_gateway_frontend::do_abort_tm_tx(
         tx = changed_tx.value();
     }
 
-    std::vector<ss::future<abort_tx_reply>> pfs;
-    for (auto rm : tx.partitions) {
-        pfs.push_back(_rm_partition_frontend.local().abort_tx(
-          rm.ntp, tx.pid, tx.tx_seq, timeout));
-    }
-    std::vector<ss::future<abort_group_tx_reply>> gfs;
-    for (auto group : tx.groups) {
-        gfs.push_back(_rm_group_proxy->abort_group_tx(
-          group.group_id, tx.pid, tx.tx_seq, timeout));
-    }
-    auto prs = co_await when_all_succeed(pfs.begin(), pfs.end());
-    auto grs = co_await when_all_succeed(gfs.begin(), gfs.end());
-    bool ok = true;
-    for (const auto& r : prs) {
-        ok = ok && (r.ec == tx_errc::none);
-    }
-    for (const auto& r : grs) {
-        ok = ok && (r.ec == tx_errc::none);
+    auto retries = _metadata_dissemination_retries;
+    auto delay_ms = _metadata_dissemination_retry_delay_ms;
+    auto done = false;
+    while (0 < retries--) {
+        std::vector<ss::future<abort_tx_reply>> pfs;
+        for (auto rm : tx.partitions) {
+            pfs.push_back(_rm_partition_frontend.local().abort_tx(
+              rm.ntp, tx.pid, tx.tx_seq, timeout));
+        }
+        std::vector<ss::future<abort_group_tx_reply>> gfs;
+        for (auto group : tx.groups) {
+            gfs.push_back(_rm_group_proxy->abort_group_tx(
+              group.group_id, tx.pid, tx.tx_seq, timeout));
+        }
+        auto prs = co_await when_all_succeed(pfs.begin(), pfs.end());
+        auto grs = co_await when_all_succeed(gfs.begin(), gfs.end());
```

Comment:
I think that @rystsov has an issue tracking this same pattern in another location (so maybe it makes sense to fix all the cases at the same time. i'd check with him about the plan). But this has a failure mode:

if `pfs` and `gfs` both contain a future that failed, then the first when_all_succeed will throw, and `gfs` will be destroyed resulting in seastar raising errors about abandoned exceptional future.

EDIT: i see it looks like this is just code being moved around and already had the issue in it. so it's probably the same thing that Denis is tracking. therefore, there probably isn't anything that needs to be done here.